src/librustc_error_codes/error_codes/E0657.md

```diff
@@ -0,0 +1,64 @@
+The `impl Trait` can only capture lifetimes bound at the fn or impl level.
+
+Erroneous code example:
+
+```compile_fail,E0657
+#![allow(warnings)]
+
+trait Id<T> {}
+trait Lt<'a> {}
+
+impl<'a> Lt<'a> for () {}
+impl<T> Id<T> for T {}
+
+fn free_fn_capture_hrtb_in_impl_trait()
+    -> Box<for<'a> Id<impl Lt<'a>>>
+{
+    Box::new(())
+}
+
+struct Foo;
+impl Foo {
+    fn impl_fn_capture_hrtb_in_impl_trait()
+        -> Box<for<'a> Id<impl Lt<'a>>>
+    {
+        Box::new(())
+    }
+}
+
+# fn main() {}
+```
+
+Here, you have used the inappropriate lifetime in the `impl Trait`,
+The `impl Trait` can only capture lifetimes bound at the fn or impl
+ level.
+
+To fix this we have to define the lifetime at the function or impl
+level and use that lifetime in the `impl Trait`. For example you can
+define the lifetime at the function.
+
+```
+#![allow(warnings)]
+
+trait Id<T> {}
+trait Lt<'a> {}
+
+impl<'a> Lt<'a> for () {}
```

Comment:
And just like you have to add `// error!`, you need to add a comment like `// ok!` on the fixed lines.